### PR TITLE
added debug flags for differentation

### PR DESCRIPTION
**Pull request by wibraun**

### CODE REVIEW

File: Compiler/Util/Flags.mo [hunk 1]
```diff
@@ -462,6 +462,10 @@ constant DebugFlag DUMP_FUNCTIONS = DEBUG_FLAG(147, "dumpFunctions", false,
   Util.gettext("Add functions to backend dumps."));
 constant DebugFlag BUILD_STATIC_SOURCE_FMU = DEBUG_FLAG(148, "buildStaticSourceFMU", false,
   Util.gettext("A temporary flag to not link the C run-time system when building an FMU; instead compiling the run-time sources into the FMU. The goal is to make this a truly static shared object, depending on nothing outside the FMU (all sources will be included)."));
+constant DebugFlag DEBUG_DIFFERENTATION = DEBUG_FLAG(149, "debugDifferentation", false,
+  Util.gettext("Dumps debug output for the differentation process."));
```
Comment (lochel) [inline Compiler/Util/Flags.mo]: Please fix the typo. It should be "Dumps debug output for the differentiation process.".

File: Compiler/Util/Flags.mo [hunk 2]
```diff
@@ -462,6 +462,10 @@ constant DebugFlag DUMP_FUNCTIONS = DEBUG_FLAG(147, "dumpFunctions", false,
...
+constant DebugFlag DEBUG_DIFFERENTATION_VERBOSE = DEBUG_FLAG(150, "debugDifferentationVerbose", false,
+  Util.gettext("Dumps verbose debug output for the differentation process."));
```
Comment (lochel) [inline Compiler/Util/Flags.mo]: Please fix the typo. It should be "Dumps verbose debug output for the differentiation process.".

File: Compiler/Util/Flags.mo [hunk 3]
```diff
@@ -616,7 +620,9 @@ constant list<DebugFlag> allDebugFlags = {
   EVAL_OUTPUT_ONLY,
   HARDCODED_START_VALUES,
   DUMP_FUNCTIONS,
-  BUILD_STATIC_SOURCE_FMU
+  BUILD_STATIC_SOURCE_FMU,
+  DEBUG_DIFFERENTATION,
+  DEBUG_DIFFERENTATION_VERBOSE
```
Comment (lochel) [inline Compiler/Util/Flags.mo]: Please fix the typos: DEBUG_DIFFERENTIATION and DEBUG_DIFFERENTIATION_VERBOSE.

File: Compiler/Util/Flags.mo (hunk 1 above)
Comment (lochel) [inline Compiler/Util/Flags.mo]: Please fix also the flag name.

File: Compiler/Util/Flags.mo (hunk 2 above)
Comment (lochel) [inline Compiler/Util/Flags.mo]: Please fix also the flag name.

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2015-10-14_16-48-21](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/944/console).

**Comment by casella** [comment]
I don't want to be pedantic, but let's try to avoid the eqdistant = true syndrome :)

**Pull request closed without merging**